libs/vkd3d/command.c

```diff
@@ -682,38 +687,39 @@ static HRESULT d3d12_fence_signal(struct d3d12_fence *fence, uint64_t physical_v
 
     /* With multiple fence workers, it is possible that signal calls are
      * out of order. The physical value is monotonic, so we can use it to
-     * check if that's the case and ignore calls with a lower value. */
-    if (fence->physical_value > physical_value)
-        return S_OK;
-
-    /* In case we have a rewind signalled from GPU, we need to recompute the max pending timeline value. */
-    new_max_pending_virtual_timeline_value = 0;
-
-    for (i = 0, j = 0; i < fence->pending_updates_count; i++)
+     * check if that's the case and ignore calls with a lower value.
+     * We don't expect the loop to run more than once, but there might be extreme edge cases where we signal
```

Comment:
Slightly weird formatting with this super long line

Reply:
Rewrote the comment.